### PR TITLE
PI-2358 custody key dates update do not ignore soft deleted records

### PR DESCRIPTION
**Pull request by achimber-moj**

### CODE REVIEW

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt [hunk 1]
```diff
@@ -48,4 +42,6 @@ class KeyDate(
             it.version = version
```
Comment (marcus-bcl) [inline projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt:42]: Don't we need to un-delete the record as well?
```suggestion
            it.version = version
            it.softDeleted = false
```

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt [hunk 2]
```diff
@@ -48,4 +42,6 @@ class KeyDate(
...
         }
     }
+
+    fun getSoftDeleteFlag() = softDeleted
```
Comment (marcus-bcl) [inline projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt:46]: Do you need this getter?

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt (hunk 1 above)
Comment (achimber-moj) [reply projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt:42]: the entity set the soft delete flag to false by default.  however, i will make the change to make this clearer

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt (hunk 2 above)
Comment (achimber-moj) [reply projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt:46]: i was using it in my test, but will delete it, i am can use the property directly

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt (hunk 1 above)
Comment (achimber-moj) [reply projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt:42]: done

File: projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt (hunk 2 above)
Comment (achimber-moj) [reply projects/custody-key-dates-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/custody/date/KeyDate.kt]: done

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**